verifier/src/lib.rs

```diff
@@ -79,11 +84,38 @@ pub enum VerificationError {
     },
 }
 
+impl DisplayableError for VerificationError {}
+
+#[derive(Debug, Clone)]
+/// A [`CtOption`] wrapped to implement the [`Display`] trait.
+pub struct DisplayableCtOption<T>(CtOption<T>);
+
+impl<T> From<CtOption<T>> for DisplayableCtOption<T> {
+    fn from(ct_option: CtOption<T>) -> Self {
+        Self(ct_option)
+    }
+}
+
+impl<T: DisplayableError> Display for DisplayableCtOption<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        let option: Option<T> = self.0.clone().into();
+        match option {
+            Some(value) => match f.alternate() {
+                true => write!(f, "{value:#}")?,
+                false => write!(f, "{value}")?,
```

Comment:
Initially I had
```suggestion
                true => write!(f, "Failure: {value:#}")?,
```
but the `Failure: ` string looked noisy in the output and I don't think "passed" is likely to get confused for an error.